apis/v1beta1/httproute_types.go

```diff
@@ -263,6 +263,55 @@ type HTTPRouteRule struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	BackendRefs []HTTPBackendRef `json:"backendRefs,omitempty"`
+
+	// Timeouts defines the timeouts that can be configured for an HTTP request.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// <gateway:experimental>
+	Timeouts *HTTPRouteTimeouts `json:"timeouts,omitempty"`
+}
+
+// HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
+// Timeout values are formatted like 1h/1m/1s/1ms as parsed by Golang time.ParseDuration
+// and MUST BE >= 1ms or 0 to disable (no timeout).
+type HTTPRouteTimeouts struct {
+	// Request specifies the duration for processing an HTTP client request after which the
+	// gateway will time out if unable to send a response.
+	//
+	// For example, setting this field to the value `10s` in an HTTPRoute will cause
+	// a timeout if a client request is taking longer than 10 seconds to complete.
+	//
+	// This timeout is intended to cover as close to the whole request-response transaction
+	// as possible although an implementation MAY choose to start the timeout after the entire
+	// request stream has been received instead of immediately after the transaction is
+	// initiated by the client.
+	//
+	// When this field is unspecified, request timeout behavior is implementation-dependent.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// +kubebuilder:validation:Pattern=^([0-9]{1,5}(h|m|s|ms)){1,4}$
+	Request *string `json:"request,omitempty"`
```

Comment:
This should use type Duration, which already includes the validation pattern.
```suggestion
	Request *Duration `json:"request,omitempty"`
```

Reply:
I think this might also be the source of your CRD generation problems - this and at least 1 other regex added in this PR are not surrounded by backticks which may be having some unintended consequences.